fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabaseFactoryImpl.java

```diff
@@ -93,6 +94,16 @@ public class FDBDatabaseFactoryImpl extends FDBDatabaseFactory {
     @Nonnull
     private Supplier<Boolean> transactionIsTracedSupplier = LOGGER::isTraceEnabled;
 
+    @API(API.Status.INTERNAL)
+    @VisibleForTesting
+    public static FDBDatabaseFactoryImpl testInstance(@Nonnull FDB initedFDB, @Nonnull APIVersion apiVersion) {
```

Comment:
This allows every test to have its own copy of the `FDBDatabaseFactory`. This means we will be able to run several concurrent tests that all access member variables of the `FDBDatabaseFactory` (which is normally a singleton). It also ensures that each test gets its own instance of an `FDBDatabase`, which means each one will independently track contexts to see if any are left un-opened by the test